shared/src/model/notifications.coffee

```diff
@@ -56,16 +61,31 @@ Notifications = {
       NOTICES = without(NOTICES, find(NOTICES, id: notice.id))
       @emit('change')
 
+  hide: (notice) ->
+    HIDDEN.push(notice)
+    @emit('change')
+
+  unhide: (notice) ->
+    HIDDEN = reject(HIDDEN, notice)
+    @emit('change')
+
+  rejectHidden: (notices) ->
+    reject(notices, (notice) ->
+      find(HIDDEN, (hidden) ->
+        isMatch(notice, hidden)
+      )
+    )
+
   # Notices originate via multiple methods.
   # * Pollers periodically check verious network endpoints
   # * Generated when the `setCourseRole` method is called
   #   based on the user's relationship with the course
   # The `getActive` method is the single point for checking which notifications
   # should be displayed
   getActive: ->
-    active = clone(NOTICES)
+    active = clone(@rejectHidden(NOTICES))
     for type, poller of POLLERS
-      active.push(clone(notice)) for notice in poller.getActiveNotifications()
+      active.push(clone(notice)) for notice in @rejectHidden(poller.getActiveNotifications())
     active
```

Comment:
Since reject creates a copy we no longer need to call `clone`  I'd only had that there so that if the caller altered the array it wouldn't alter the private `NOTICES`